projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php

```diff
@@ -173,6 +178,39 @@ public function set_site_card_menu_class( array $menu ) {
 		return $menu;
 	}
 
+	/**
+	 * Returns the first available upsell nudge.
+	 *
+	 * @return array
+	 */
+	public function get_upsell_nudge() {
+		require_lib( 'jetpack-jitm/jitm-engine' );
+		$jitm_engine = new JITM\Engine();
+
+		$message_path = 'calypso:sites:sidebar_notice';
+		$current_user = wp_get_current_user();
+		$user_id      = $current_user->ID;
+		$user_roles   = implode( ',', $current_user->roles );
+		$query_string = array(
+			'message_path' => $message_path,
+		);
+
+		// Get the top message only.
+		$message = $jitm_engine->get_top_messages( $message_path, $user_id, $user_roles, $query_string )[0];
```

Comment:
Not blocking: $jitm_engine->get_top_messages( $message_path, $user_id, $user_roles, $query_string **)[0];** this will trigger a warning in PHP 8 and above if the key does not exist. :)

Reply:
Done e58d6f5